### ISSUE TITLE
Boolean evaluation bug in J2ObjC

### ISSUE THREAD

**Issue by yegork78** [google/j2objc#1878]
Consider this simple class:

```
class Test
{
  boolean test()
  {
    return true == true == true;
  }
}
```

It will result in "AssertionError: null" failure upon running `./j2objc Test.java`.
A workaround is to replace the boolean expression `true == true == true` with `(true == true) == true` (even though the original is a valid Java code, working in Android and with GWT).

**Comment by tomball** [comment]
The assertion that's failing is verifying that there are two operands in the equalsequals expression, and now we know that's false. So I think the right fix is to generate a new infix expression with the first two operands, and use it as the first operand in another infix expression (this will work with additional operands). We won't need to use a parenthesized expression, though, because >2 operand infix expressions are also legal in Objective-C.

**Comment by tomball** [comment]
Fixed in [655aab4](https://github.com/google/j2objc/commit/655aab45e17deb5d859435fc3e4cf8d67e62e80e).
